### ISSUE TITLE
About the paradigm of code-BERT training

### ISSUE THREAD

**Issue by Sewens** [microsoft/CodeBERT#1]
The paper has mentioned that the model is inspired by ELECTRA which combining contrastive learning in programming LM.

While in [run_classifier::train](https://github.com/microsoft/CodeBERT/blob/47e5f6146ea4271d143940930904b1f68ab8e45a/codesearch/run_classifier.py#L53) I can not find the generator and discriminator part.

These four files only describe how to train BERT model (in the context Robert) on the source code corpus.

I'm feeling confused about the training process, pls help.

**Comment by fengzhangyin** [comment]
Yes, we use masked language modeling objective and replaced token detection objective to pre-train CodeBERT.

As stated in the paper, in the fine-tuning step, we only use discriminator to produce general-purpose representations.

Each subdirectory corresponds to the fine-tuning source code of a downstream task. We release the pre-trained model and the source code of downstream tasks, not including the source code of the pre-training step.

**Comment by Sewens** [comment]
Will there be a further plan to release the pre-training sources?

**Comment by fengzhangyin** [comment]
Sorry, we don't have this plan.

**Comment by Sewens** [comment]
Okay, thank you for your patience :)
